papirus/textpos.py

```diff
@@ -107,24 +107,25 @@ def addToImageText(self, Id, font_path='/usr/share/fonts/truetype/freefont/FreeM
         # Go through the lines and add them
         for line in toProcess:
             # Add in a line to add the words to
-            text_lines.append("")
+            text_lines.append(u"")
+            current_line += 1
             # Compute each line
             for word in line.split():
-                # If there is space on line add the word to it
-                if (len(text_lines[current_line]) + len(word)) < line_size:
-                    # Only add a space if there`s something on the line
-                    if len(text_lines[current_line]) > 0:
-                        text_lines[current_line] += " "
+                # Always add first word (even it is too long)
+                if len(text_lines[current_line]) == 0:
                     text_lines[current_line] += word
+                elif (x + draw.textsize(text_lines[current_line] + " " + word, font=font)[0]) < line_width:
+                    text_lines[current_line] += " " + word
                 else:
                     # No space left on line so move to next one
-                    text_lines.append("")
+                    text_lines.append(u"")
                     current_line += 1
                     text_lines[current_line] += " " + word
```

Comment:
I believe this should be: `text_lines[current_line] += word`   There is no need to add the space before as you have incremented the line.